### PR TITLE
Final Capstone Project: Book an Appointment.

### PR DESCRIPTION
**Pull request by Reem-lab**
**This project is separated into two repositories: [Front End](https://github.com/piero-vic/house-booking-front-end) (holds the React App) and [Back End](https://github.com/Reem-lab/house-booking-back-end) (holds the API).**

Please also check the [PR for the front-end](https://github.com/piero-vic/house-booking-front-end/pull/16).

**IMPORTANT: We have a `credentials.yml.enc` file. We're using `secret_key_base` for JWT. We're sending the master key in the submission form. You need to create a file called `master.key` containing the key that we're sending in order for the app to run on your local machine.**

## Changes that we made here As a team 💪 🍄 
 - Set up the Kanban board with all the cards for tasks.
 - Specify the Topic as **a Renting a house.**
 - Set up the `Gitflow`.
 - Set up the `rails` app with `Postgresql`
 - Set up the authentication and authorization with Devise JWT.
 - Set up the linters files
 - Create the models `user` and `house` and `reservations`.
 - Create All migrations needed and schema
 - Create the associations.
 - Add the Validations.
 - Create the routes For `Login` and `Signup` and `Logout`
 - Create the routes for our pages `Houses` and `reservations`
 - Create controller actions for houses.
 - Create controller actions for reservations.
 - Add tests for controllers.
 - Add tests for models.
 - Deploy the API with `Heroku`
 - Create the Documentation for the API by `postman`.
 - Add the link for the documentation link in the `README` file.
 - Add a link for the Front-end repository in the `README` file
 - Add the `README.md` file
 - Add the `MIT.md` file
 - No linters error. 🚀💯🔥

### CODE REVIEW

**Review by vichuge — CHANGES_REQUESTED** [review]
Hi dear student, 

Good job so far!
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!

mario_ladders


## Suggested changes
_Check the comments under the review._

- I'm afraid the test on FE doesn't exist, please add it 🐶 
image

- Once I try to install the project locally, when I tried to use `rails db:create` after installing the gems, I got this issue, I could think this was something about my computer, but I just check another project and everything were good. To recreate this issue, please delete all you got in your local machine, and then install everything from scratch. Maybe is my mistake, if that is the case, ignore this. 🐱 
image

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

### Good job on the design of the project, I could see how to work hard to get exactly the same style as required, good job!

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  get '/v1/current_user', to: 'v1/users/current_user#index'
+
+  devise_for :users, path: '', path_names: {
+    sign_in: 'v1/login',
+    sign_out: 'v1/logout',
+    registration: 'v1/signup'
+  },
+  controllers: {
+    sessions: 'v1/users/sessions',
+    registrations: 'v1/users/registrations'
```
[the remark this answers is not in the document]
Comment (piero-vic) [reply config/routes.rb:11]: Can you give us more details? Is there another way we can authenticate the user?

**Comment by piero-vic** [comment]
> * Once I try to install the project locally, when I tried to use `rails db:create` after installing the gems, I got this issue, I could think this was something about my computer, but I just check another project and everything were good. To recreate this issue, please delete all you got in your local machine, and then install everything from scratch. Maybe is my mistake, if that is the case, ignore this. cat
>   image

We have a `credentials.yml.enc` file. Maybe we forgot to send you the master key. We're using `secret_key_base` for JWT. One thing you can do to try the app locally is delete the `credentials.yml.enc` file inside `config` folder and generate a new one.

File: config/routes.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (vichuge) [reply config/routes.rb:11]: After having the zoom call we got a few minutes ago, please ignore this issue, it was my mistake. 😄

**Review by MuriloRoque — APPROVED** [review]
Hi @Reem-lab, @piero-vic, @aimemalaika, @Sahar-AbdelSamad, @German-Cobian, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**